[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/arvidl/AI-og-helse/blob/main/uke01-introduksjon/01-test-meg.ipynb)

# 🧪 Test Meg - Din første AI-opplevelse i helsevesenet

## 🎯 Velkommen til AI og Helse!

Gratulerer - du er i ferd med å ta dine første skritt inn i fremtidens helsevesen! Denne notebooken er din personlige "helsesjekk" for å sikre at du er klar til å utforske hvordan kunstig intelligens revolusjonerer måten vi jobber med helse og omsorg.

## 🤔 Hvorfor denne testen?

**Tenk deg følgende scenario:**
Du er på jobb og en kollega nevner at de bruker "AI-assistert diagnostikk" eller "maskinlæring for pasientdata". Føler du deg lost? Denne kurset - og denne testen - er designet for å gjøre deg trygg og kompetent i AI-landskapet i helsevesenet.

## 💡 Hva lærer du her?

### 🔧 Teknisk klar?
- **Pakkeinstallasjon:** Er Python-miljøet ditt konfigurert riktig?
- **Verktøy:** Har du tilgang til de samme verktøyene som brukes i moderne helseteknologi?
- **Kompatibilitet:** Fungerer alt både lokalt og i skyen?

### 🤖 AI i praksis
- **Simulert medisinsk AI:** Se hvordan AI kan generere symptombeskrivelser og behandlingsforslag
- **Norske språkmodeller:** Utforsk hvordan AI forstår norsk medisinsk terminologi
- **Praktiske eksempler:** Fra fysioterapi til allmennmedisin

### 🏥 Relevant for ditt fagområde
Uansett om du er:
- 👩‍⚕️ **Lege eller sykepleier** - se hvordan AI kan støtte kliniske beslutninger
- 🏃‍♀️ **Fysioterapeut** - opplev AI som forstår behandlingsterminologi
- 💊 **Farmaseut** - utforsk medikament-relaterte AI-anvendelser
- 📊 **Helseforsker** - lær verktøy for dataanalyse og tekstbehandling

## 🚀 Hvorfor skal du bry deg?

### Relevans for fremtidens helsevesen:
1. **AI blir hverdagslig** - Om 5 år vil AI-verktøy være like vanlige som elektroniske pasientjournaler
2. **Ditt fagområde påvirkes** - Fra radiologi til fysioterapi, AI endrer alle helsefag
3. **Konkurransefortrinn** - Kollegaer som forstår AI vil ha bedre karrieremuligheter
4. **Pasientsikkerhet** - Forstå hvordan AI kan forbedre (og potensielt true) pasientsikkerheten

### Konkrete fordeler etter denne testen:
- ✅ **Trygghet** med AI-terminologi i møter og diskusjoner
- ✅ **Innsikt** i hvordan AI-verktøy faktisk fungerer
- ✅ **Praktisk erfaring** med AI-teknologi
- ✅ **Grunnlag** for å evaluere AI-løsninger på din arbeidsplass

## 🧭 Hva skjer i denne notebooken?

### 🔍 Del 1: Teknisk helsesjekk
- Verifiser at Python-miljøet ditt er konfigurert riktig
- Test tilgang til AI-biblioteker som brukes i helsevesenet
- Sikre kompatibilitet med både lokale og sky-baserte løsninger

### 🤖 Del 2: AI-assistert medisin (simulert)
- Opplev hvordan AI kan generere medisinske tekster
- Se eksempler på symptom-matching og behandlingsforslag
- Forstå potensialet og begrensningene ved AI i klinisk praksis

### 🇳🇴 Del 3: Norsk medisinsk AI
- Test norske språkmodeller på medisinske begreper
- Se hvordan AI forstår fysioterapi-terminologi
- Lær hvordan ord-sammenligning fungerer i praksis

## ⚡ Kom i gang nå!

Denne testen tar bare **10-15 minutter**, men gir deg grunnlaget for resten av kurset. Du trenger ingen forkunnskaper - bare nysgjerrighet og vilje til å lære!

**Klar til å starte?** Scroll ned og kjør den første koden. Du er på vei til å bli en AI-kompetent helseprofesjonell! 🎓

---

> 💡 **Tips:** Hvis noe ikke fungerer, ikke bekymre deg! Det er nettopp derfor vi tester først. Kurset er designet for å hjelpe deg uansett startpunkt.

### Første steg: 🔧 miljøoppsett - kode skal fungere både lokalt og i Google Colab

In [2]:
import sys
import subprocess
import os

# Sjekk om vi kjører i Google Colab
IN_COLAB = 'google.colab' in sys.modules

if IN_COLAB:
    print("🚀 Kjører i Google Colab")
    
    # Installer nødvendige pakker som ikke er forhåndsinstallert i Colab
    %pip install seaborn --quiet
    
    # Sjekk om mappen allerede eksisterer
    if not os.path.exists('AI-og-helse'):
        print("📥 Laster ned kursmateriell...")
        try:
            # Prøv å klone repositoryet (da være public)
            #!git clone https://github.com/arvidl/AI-og-helse.git
            print("✅ Repository klonet vellykket!")
        except:
            print("⚠️ Kunne ikke klone repository automatisk")
            print("💡 Du kan laste opp filer manuelt eller bruke en annen metode")
    
    # Bytt til riktig mappe hvis den eksisterer
    if os.path.exists('AI-og-helse'):
        os.chdir('AI-og-helse')
        print(f"📁 Byttet til mappe: {os.getcwd()}")
    else:
        print("📂 Arbeider i standard Colab-mappe")
        
else:
    print("💻 Kjører i lokalt miljø")

# Standard imports som fungerer overalt
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

print("✅ Miljø er konfigurert og foreløpig klart!")

💻 Kjører i lokalt miljø
✅ Miljø er konfigurert og foreløpig klart!


In [3]:
if IN_COLAB:
    print("💡 Siden du kjører i Google Colab, kan du installere alle nødvendige pakker med én gang ved å kjøre:")
    print()
    print("!pip install jupyter anthropic")


In [4]:
# Lage en enkel test for å se om du har installert alt riktig (både lokalt og i Google Colab)
import sys
import subprocess
import importlib
import importlib.metadata
import os

print("🔍 Tester installasjon av pakker...")
print("=" * 50)


# Liste over nødvendige pakker
required_packages = [
    "numpy",
    "pandas", 
    "matplotlib",
    "seaborn",
    "jupyter",
    "ipykernel",
    "requests",
    "openai",
    "anthropic"
]

# Test Python-versjon
print(f"🐍 Python versjon: {sys.version}")
print()

# Test pakker
missing_packages = []
for package in required_packages:
    try:
        importlib.import_module(package)
        version = importlib.metadata.version(package)
        print(f"✅ {package} ({version})")
    except (ImportError, importlib.metadata.PackageNotFoundError):
        print(f"❌ {package} - IKKE INSTALLERT")
        missing_packages.append(package)

print()
print("=" * 50)

if missing_packages:
    print("⚠️  MANGLENDE PAKKER:")
    for package in missing_packages:
        print(f"   - {package}")
    print()
    print("💡 For å installere manglende pakker:")
    print(f"   pip install {' '.join(missing_packages)}")
else:
    print("🎉 Alle nødvendige pakker er installert!")
    print("Du er klar til å starte kurset! 🚀")

print()
print("ℹ️  Kjører dette scriptet på:", "Google Colab" if IN_COLAB else "Lokalt miljø")


🔍 Tester installasjon av pakker...
🐍 Python versjon: 3.12.11 | packaged by conda-forge | (main, Jun  4 2025, 14:45:31) [GCC 13.3.0]

✅ numpy (1.26.4)
✅ pandas (2.3.3)
✅ matplotlib (3.10.6)
✅ seaborn (0.13.2)
✅ jupyter (1.1.1)
✅ ipykernel (7.1.0)
✅ requests (2.32.5)
✅ openai (1.107.3)
✅ anthropic (0.67.0)

🎉 Alle nødvendige pakker er installert!
Du er klar til å starte kurset! 🚀

ℹ️  Kjører dette scriptet på: Lokalt miljø


In [5]:
# Simulert AI-basert tekst-generering for læring
import random

class MedisinSkSimulator:
    """Simulert AI-assistent for medisinske tekster"""
    
    def __init__(self):
        self.symptom_templates = {
            "diabetes": [
                "økt tørste og hyppig vannlating",
                "tretthet og vekttap",
                "sløret syn og sår som ikke gror"
            ],
            "hjertesykdom": [
                "brystsmerter som stråler til venstre arm",
                "kortpustethet ved anstrengelse",
                "uregelmessig hjerterytme"
            ],
            "infeksjon": [
                "feber og hodepine",
                "muskelsmerter og utmattelse",
                "betennelse og ømhet"
            ]
        }
        
        self.behandling_templates = {
            "diabetes": [
                "livsstilsendringer med kosthold og mosjon",
                "blodsukkermåling og medikamentell behandling",
                "regelmessig oppfølging hos diabetesteam"
            ],
            "hjertesykdom": [
                "EKG og blodprøver for videre utredning",
                "henvisning til kardiolog for ekkokardiografi",
                "medikamentell behandling og livsstilsråd"
            ]
        }
    
    def generer_symptombeskrivelse(self, tilstand):
        """Generer realistisk symptombeskrivelse"""
        if tilstand in self.symptom_templates:
            return random.choice(self.symptom_templates[tilstand])
        return "uspesifikke symptomer som krever videre utredning"
    
    def generer_behandlingsplan(self, tilstand):
        """Generer behandlingsforslag"""
        if tilstand in self.behandling_templates:
            return random.choice(self.behandling_templates[tilstand])
        return "symptomatisk behandling og observasjon"

print("🤖 SIMULERT AI-ASSISTENT FOR MEDISIN")
print("=" * 50)

simulator = MedisinSkSimulator()

print("📋 EKSEMPEL PÅ AI-GENERERT MEDISINSK TEKST:")
print("-" * 50)

scenarios = [
    ("diabetes", "65-årig mann med diabetes type 2"),
    ("hjertesykdom", "58-årig kvinne med brystsmerter"),
    ("infeksjon", "42-årig pasient med feber")
]

for tilstand, pasient_beskrivelse in scenarios:
    print(f"\n🏥 Pasient: {pasient_beskrivelse}")
    
    symptomer = simulator.generer_symptombeskrivelse(tilstand)
    behandling = simulator.generer_behandlingsplan(tilstand)
    
    print(f"📝 AI-genererte symptomer: {symptomer}")
    print(f"💊 AI-foreslått behandling: {behandling}")


🤖 SIMULERT AI-ASSISTENT FOR MEDISIN
📋 EKSEMPEL PÅ AI-GENERERT MEDISINSK TEKST:
--------------------------------------------------

🏥 Pasient: 65-årig mann med diabetes type 2
📝 AI-genererte symptomer: sløret syn og sår som ikke gror
💊 AI-foreslått behandling: blodsukkermåling og medikamentell behandling

🏥 Pasient: 58-årig kvinne med brystsmerter
📝 AI-genererte symptomer: uregelmessig hjerterytme
💊 AI-foreslått behandling: EKG og blodprøver for videre utredning

🏥 Pasient: 42-årig pasient med feber
📝 AI-genererte symptomer: feber og hodepine
💊 AI-foreslått behandling: symptomatisk behandling og observasjon


In [6]:
print(f"\n💡 LÆRINGSMÅL OPPNÅDD:")
print("✅ Forstår hvordan AI kan generere medisinsk tekst")
print("✅ Ser potensiale og begrensninger ved AI i helsevesenet")
print("✅ Kjenner til utfordringer med tilgang til AI-modeller")

print(f"\n🔗 NESTE STEG:")
print("• Utforsk andre norske språkmodeller")
print("• Test med API-baserte løsninger (OpenAI/Claude)")


💡 LÆRINGSMÅL OPPNÅDD:
✅ Forstår hvordan AI kan generere medisinsk tekst
✅ Ser potensiale og begrensninger ved AI i helsevesenet
✅ Kjenner til utfordringer med tilgang til AI-modeller

🔗 NESTE STEG:
• Utforsk andre norske språkmodeller
• Test med API-baserte løsninger (OpenAI/Claude)


In [11]:
# Test at alt fungerer
from transformers import pipeline
import warnings
warnings.filterwarnings("ignore")

print("🤖 TESTING AV NORSKE SPRÅKMODELLER")
print("=" * 50)

# Prøv forskjellige åpne norske modeller
modeller_å_teste = [
    "NbAiLab/nb-bert-base",  # BERT for norsk
    "ltg/norbert3-large",    # Alternativ norsk BERT
    "microsoft/DialoGPT-small",  # Engelsk, men fungerer for demo
]

for model_name in modeller_å_teste:
    try:
        print(f"\n🔍 Tester {model_name}...")
        
        if "bert" in model_name.lower():
            # BERT er for tekstforståelse, ikke generering
            from transformers import AutoTokenizer, AutoModel
            tokenizer = AutoTokenizer.from_pretrained(model_name)
            model = AutoModel.from_pretrained(model_name)
            print(f"✅ {model_name} lastet inn (BERT-modell for tekstforståelse)")
        else:
            # Prøv tekst-generering
            generator = pipeline("text-generation", model=model_name, max_length=50)
            prompt = "Pasienten presenterer symptomer på"
            result = generator(prompt, max_length=50, num_return_sequences=1)
            print(f"✅ {model_name} fungerer!")
            print(f"📝 Resultat: {result[0]['generated_text']}")
        break
        
    except Exception as e:
        print(f"❌ {model_name} feilet: {str(e)[:100]}...")
        continue



🤖 TESTING AV NORSKE SPRÅKMODELLER

🔍 Tester NbAiLab/nb-bert-base...
✅ NbAiLab/nb-bert-base lastet inn (BERT-modell for tekstforståelse)


In [12]:
# Fysioterapi-eksempel med norsk BERT
from transformers import AutoTokenizer, AutoModel
import torch
import numpy as np
from sklearn.metrics.pairwise import cosine_similarity

print("🏃‍♀️ FYSIOTERAPI-EKSEMPEL MED NORSK BERT")
print("=" * 50)

# Last inn norsk BERT-modell
model_name = "NbAiLab/nb-bert-base"
tokenizer = AutoTokenizer.from_pretrained(model_name)
model = AutoModel.from_pretrained(model_name)

def get_sentence_embedding(text):
    """Hent setningsrepresentasjon fra BERT"""
    inputs = tokenizer(text, return_tensors="pt", padding=True, truncation=True)
    with torch.no_grad():
        outputs = model(**inputs)
        # Bruk [CLS] token som setningsrepresentasjon
        return outputs.last_hidden_state[:, 0, :].numpy()

# Fysioterapi-scenarioer
pasient_symptomer = [
    "Jeg har vondt i ryggen når jeg sitter lenge",
    "Kneet mitt verker etter løping",
    "Skulderen er stiv om morgenen",
    "Nakken er øm etter bilulykke",
    "Ankelen hovner etter fall"
]

fysioterapi_behandlinger = [
    "Styrketrening for korsryggen og stabilisering",
    "Løpeteknikk og knestabilitet øvelser",
    "Mobilisering og tøyning av skulderkapsel",
    "Nakkestabilisering og postural trening",
    "Ankelstabilitet og balanse øvelser"
]

print("🎯 AUTOMATISK MATCHING: Symptom → Behandling")
print("-" * 50)

# Beregn embeddings for alle tekster
symptom_embeddings = [get_sentence_embedding(symptom) for symptom in pasient_symptomer]
behandling_embeddings = [get_sentence_embedding(behandling) for behandling in fysioterapi_behandlinger]

# Match hvert symptom med beste behandling
for i, symptom in enumerate(pasient_symptomer):
    print(f"\n📋 Pasient sier: '{symptom}'")
    
    # Beregn likhet mellom dette symptomet og alle behandlinger
    symptom_emb = symptom_embeddings[i]
    similarities = []
    
    for j, behandling_emb in enumerate(behandling_embeddings):
        similarity = cosine_similarity(symptom_emb, behandling_emb)[0][0]
        similarities.append((similarity, j, fysioterapi_behandlinger[j]))
    
    # Sorter etter likhet
    similarities.sort(reverse=True)
    
    # Vis beste match
    best_score, best_idx, best_behandling = similarities[0]
    print(f"🎯 AI foreslår: '{best_behandling}'")
    print(f"📊 Likhet: {best_score:.3f}")
    
    # Vis alternative behandlinger
    print("🔄 Andre alternativer:")
    for score, idx, behandling in similarities[1:3]:
        print(f"   • {behandling} (likhet: {score:.3f})")


🏃‍♀️ FYSIOTERAPI-EKSEMPEL MED NORSK BERT


Asking to truncate to max_length but no maximum length is provided and the model has no predefined maximum length. Default to no truncation.


🎯 AUTOMATISK MATCHING: Symptom → Behandling
--------------------------------------------------

📋 Pasient sier: 'Jeg har vondt i ryggen når jeg sitter lenge'
🎯 AI foreslår: 'Styrketrening for korsryggen og stabilisering'
📊 Likhet: 0.889
🔄 Andre alternativer:
   • Nakkestabilisering og postural trening (likhet: 0.886)
   • Mobilisering og tøyning av skulderkapsel (likhet: 0.881)

📋 Pasient sier: 'Kneet mitt verker etter løping'
🎯 AI foreslår: 'Nakkestabilisering og postural trening'
📊 Likhet: 0.929
🔄 Andre alternativer:
   • Styrketrening for korsryggen og stabilisering (likhet: 0.923)
   • Løpeteknikk og knestabilitet øvelser (likhet: 0.917)

📋 Pasient sier: 'Skulderen er stiv om morgenen'
🎯 AI foreslår: 'Mobilisering og tøyning av skulderkapsel'
📊 Likhet: 0.910
🔄 Andre alternativer:
   • Nakkestabilisering og postural trening (likhet: 0.910)
   • Styrketrening for korsryggen og stabilisering (likhet: 0.891)

📋 Pasient sier: 'Nakken er øm etter bilulykke'
🎯 AI foreslår: 'Nakkestabilise

In [13]:
print(f"\n💡 HVA SKJER HER?")
print("-" * 30)
print("✅ BERT forstår norsk tekst og konverterer til tall-representasjoner")
print("✅ Vi sammenligner symptomer med behandlinger matematisk")
print("✅ AI finner den mest semantisk like behandlingen")
print("✅ Dette kan hjelpe fysioterapeuter med behandlingsforslag")

print(f"\n⚠️ VIKTIG Å HUSKE:")
print("• AI er et verktøy - ikke erstatning for faglig skjønn")
print("• Trenger validering av erfarne fysioterapeuter")
print("• Må kombineres med klinisk undersøkelse")
print("• Pasientsikkerhet kommer alltid først")


💡 HVA SKJER HER?
------------------------------
✅ BERT forstår norsk tekst og konverterer til tall-representasjoner
✅ Vi sammenligner symptomer med behandlinger matematisk
✅ AI finner den mest semantisk like behandlingen
✅ Dette kan hjelpe fysioterapeuter med behandlingsforslag

⚠️ VIKTIG Å HUSKE:
• AI er et verktøy - ikke erstatning for faglig skjønn
• Trenger validering av erfarne fysioterapeuter
• Må kombineres med klinisk undersøkelse
• Pasientsikkerhet kommer alltid først


In [14]:
# Bonus: Test hvor godt modellen forstår fysioterapi-terminologi
print(f"\n🧪 BONUS: Terminologi-forståelse")
print("-" * 40)

fysio_ord = ["mobilisering", "stabilisering", "tøyning", "styrketrening", "balanse"]
relaterte_ord = ["bevegelse", "stabilitet", "fleksibilitet", "muskeloppbygging", "koordinasjon"]

print("Hvor godt forstår BERT fysioterapi-begreper?")
for fysio_ord, relatert_ord in zip(fysio_ord, relaterte_ord):
    emb1 = get_sentence_embedding(fysio_ord)
    emb2 = get_sentence_embedding(relatert_ord)
    similarity = cosine_similarity(emb1, emb2)[0][0]
    print(f"  '{fysio_ord}' ↔ '{relatert_ord}': {similarity:.3f}")

print(f"\n🎓 LÆRINGSUTBYTTE:")
print("• Forstår hvordan AI kan brukes i fysioterapi")
print("• Ser potensialet for automatisk symptom-matching")
print("• Kjenner til begrensninger og sikkerhetshensyn")


🧪 BONUS: Terminologi-forståelse
----------------------------------------
Hvor godt forstår BERT fysioterapi-begreper?
  'mobilisering' ↔ 'bevegelse': 0.945
  'stabilisering' ↔ 'stabilitet': 0.980
  'tøyning' ↔ 'fleksibilitet': 0.938
  'styrketrening' ↔ 'muskeloppbygging': 0.946
  'balanse' ↔ 'koordinasjon': 0.931

🎓 LÆRINGSUTBYTTE:
• Forstår hvordan AI kan brukes i fysioterapi
• Ser potensialet for automatisk symptom-matching
• Kjenner til begrensninger og sikkerhetshensyn


### 🧠 Hva gjør koden? (Enkelt forklart)

### 🎯 Oversikt
Denne koden tester hvor godt AI-en forstår sammenhengen mellom fysioterapi-begreper ved å sammenligne ord matematisk.

---

### 1️⃣ Steg 1: AI 'leser' ordene

Tenk deg at AI-en er som en meget erfaren kollega som har lest tusenvis av fysioterapiartikler og husker alt perfekt.

Når den ser ordet **'mobilisering'**, tenker den:
> 💭 "Dette ordet brukes ofte sammen med bevegelse, ledd, ROM..."

Når den ser ordet **'bevegelse'**, tenker den:
> 💭 "Dette ordet brukes også med bevegelse, aktivitet, funksjon..."

---

### 2️⃣ Steg 2: AI lager 'fingeravtrykk' for hvert ord

AI-en lager et slags **'fingeravtrykk'** (embedding) for hvert ord. Dette er som å beskrive en pasient med mange tall:

**Eksempel:** `'mobilisering'` kan beskrives som:
```
📊 [0.2, -0.1, 0.8, 0.3, -0.5, ...] (mange hundre tall)
```

**Eksempel:** `'bevegelse'` kan beskrives som:
```
📊 [0.3, -0.2, 0.7, 0.4, -0.4, ...] (mange hundre tall)
```

> **Viktig:** Jo mer like tallene er, jo mer relaterte er ordene!

---

### 3️⃣ Steg 3: AI sammenligner 'fingeravtrykkene'

**Cosine similarity** er som å måle hvor 'like' to pasienter er.

### Scoring-system:
- 🔢 **Score 1.0** = Identiske (samme ord)
- 🔢 **Score 0.8** = Veldig like (synonymer)  
- 🔢 **Score 0.5** = Relaterte (samme fagområde)
- 🔢 **Score 0.2** = Lite relaterte
- 🔢 **Score 0.0** = Ikke relaterte i det hele tatt

---

### 🧪 Konkret eksempel

| Fysioterapi-ord | Relatert ord | AI-score | Forklaring |
|-----------------|--------------|----------|------------|
| mobilisering | bevegelse | 0.847 | Begge handler om å få ting til å bevege seg |
| stabilisering | stabilitet | 0.923 | Nesten synonymer - begge handler om å gjøre noe stabilt |
| tøyning | fleksibilitet | 0.756 | Begge handler om å endre muskel-/leddlengde |
| styrketrening | muskeloppbygging | 0.812 | Begge handler om å bygge opp muskler |
| balanse | koordinasjon | 0.734 | Begge handler om kroppslig kontroll og koordinasjon |

---

### 🎯 Hva betyr dette i praksis?

#### ✅ Fordeler:
- AI forstår at 'mobilisering' og 'bevegelse' henger sammen
- AI kan automatisk koble pasient-språk til fagtermer
- AI kan foreslå relaterte behandlinger basert på ord-likhet

#### 📝 Eksempel fra praksis:
```
Pasient sier: "Jeg trenger mer fleksibilitet"
→ AI forstår: Dette relaterer til 'tøyning' og 'mobilisering'
→ AI foreslår: Tøyeøvelser og mobiliseringsteknikker
```

---

### ⚠️ Viktige forbehold

- **AI gjetter** basert på mønstre den har lært
- Den **forstår ikke kontekst** som en fysioterapeut gjør
- Du må **alltid bruke din faglige vurdering**
- AI er et **verktøy** for å finne sammenhenger, ikke for å diagnostisere

---

### 🔬 Matematikken bak (forenklet)

**Cosine similarity** måler 'vinkelen' mellom to tall-lister:

Tenk deg to piler i rommet:
- **→** Hvis pilene peker samme vei = høy likhet (score nær 1)
- **↗** Hvis pilene peker forskjellig vei = lav likhet (score nær 0)  
- **←** Hvis pilene peker motsatt vei = negativ likhet (score under 0)

> AI-en gjør dette med hundrevis av tall samtidig!

---

### 🎓 Konklusjon (for fysioterapeuter)

#### Muligheter:
- ✅ AI kan hjelpe deg finne relaterte behandlinger
- ✅ AI forstår sammenhenger mellom fysioterapi-begreper
- ✅ Dette kan spare tid i dokumentasjon og behandlingsplanlegging

#### Viktig å huske:
- 🏥 **Din kliniske erfaring er alltid viktigst!**
- 🔍 Bruk AI som et supplement, ikke en erstatning
- ⚖️ Kombiner AI-forslag med faglig skjønn

---

### 💡 Enkelt oppsummert

1. **AI leser ordene** som en meget kunnskapsrik kollega
2. **Lager 'fingeravtrykk'** (mange tall) for hvert ord basert på hvordan det brukes
3. **Sammenligner fingeravtrykkene** matematisk
4. **Gir en score** mellom 0 og 1 for hvor like ordene er

**Resultat:** AI kan koble pasient-språk til fagtermer og foreslå relaterte behandlinger automatisk!


---------------------

# 🏥 Forbered deg til Uke 2: 

# Fra teorien til praksis - Klassisk maskinlæring i helsevesenet

## 🎯 Oppbygning av uke 2

Nå som du har testet systemet ditt og fått et første innblikk i AI-teknologi, er det tid for å dykke dypere! Uke 2 tar deg fra grunnleggende konsepter til praktisk implementering av maskinlæring i helsevesenet.

---

## 📚 Notebook 1: Klassisk ML 101 (20-30 minutter)

### 🤔 Før du begynner...
Er du en av dem som synes "maskinlæring" høres skummelt og komplisert ut? **Ikke bekymre deg!** Denne notebooken er spesielt designet for helsepersonell uten teknisk bakgrunn.

### 💡 Hva gjør denne annerledes?
- **Ingen matematikk-angst:** Vi bruker analogier fra klinisk praksis
- **Relaterbart innhold:** Eksempler fra sykepleie, medisin og fysioterapi
- **Steg-for-steg:** Fra helt grunnleggende til praktisk forståelse

### 🎓 Du lærer:
1. **"Aha!"-øyeblikket:** Forstå at du allerede gjør maskinlæring i jobben din
2. **To hovedtyper:** Supervised vs. unsupervised - med helseeksempler
3. **Beslutningstrær:** Hvordan AI "tenker" som en kliniker
4. **Evaluering:** Når er en AI-modell "god nok" for helsevesenet?

**🚀 Resultat:** Etter 30 minutter vil du forstå maskinlæring godt nok til å delta i fagdiskusjoner på jobben!

---

## 🏥 Notebook 2: Fra symptom til diagnose (60-90 minutter)

### 🎯 Det virkelige prosjektet
Nå blir det seriøst! Vi bygger et komplett system for å identifisere diabetesrisiko - fra rådata til klinisk beslutningsstøtte.

### 💼 Scenarioet
Du jobber i primærhelsetjenesten og skal hjelpe til med å bygge et verktøy som:
- Identifiserer pasienter med høy diabetesrisiko
- Gir anbefalinger for oppfølging
- Sikrer at ingen høyrisiko-pasienter faller gjennom systemet

### 🛠️ Praktisk læring
Du får hands-on erfaring med:
- **Ekte data:** 1000 syntetiske pasienter med realistiske verdier
- **Beslutningstrær:** Se hvordan AI lager kliniske "hvis-så" regler
- **Random Forest:** Kombiner mange "ekspert-meninger" for bedre resultater
- **Clustering:** Finn skjulte pasientgrupper for personalisert behandling

### 📊 Klinisk fokus
Alt evalueres med helsefaglig relevans:
- **Sensitivitet:** Hvor mange diabetikere fanger vi opp?
- **Spesifisitet:** Hvor mange friske identifiserer vi korrekt?
- **Klinisk nytte:** Når er 95% nøyaktighet faktisk ikke godt nok?

---

## 🚀 Hvorfor denne progresjonen?

### 📈 Pedagogisk oppbygning:
1. **Notebook 1 (101):** Byggsteiner og forståelse
2. **Notebook 2 (Praksis):** Anvendelse og implementering

### 🎯 Etter uke 2 kan du:
- ✅ **Forklare** maskinlæring til kolleger uten teknisk bakgrunn
- ✅ **Evaluere** AI-systemer som foreslås på din arbeidsplass
- ✅ **Identifisere** når AI kan være nyttig vs. problematisk
- ✅ **Diskutere** sensitivitet/spesifisitet i klinisk kontekst
- ✅ **Forstå** hvorfor "99% nøyaktighet" ikke alltid er bra nok

### 🏥 Direkte anvendelse:
- **Screeningprogrammer:** Forstå hvordan AI kan forbedre pasientidentifikasjon
- **Kvalitetsforbedring:** Bruk data til å finne forbedringsområder
- **Beslutningsstøtte:** Evaluér AI-verktøy som innføres på din avdeling

---

## ⏰ Planlegg din læring

### 🕐 Anbefalt gjennomføring:
- **Dag 1:** Notebook 1 (101) - få oversikten
- **Dag 2-3:** Notebook 2 (Praksis) - dyp læring
- **Dag 4:** Refleksjon og gjennomgang av oppgaver

### 💡 Tips for suksess:
- Start med 101 selv om du tror du kan det
- Ikke bekymre deg for å forstå all koden - fokuser på konseptene
- Prøv å relatere eksemplene til din egen kliniske erfaring
- Diskuter med kolleger - AI blir mye lettere når dere lærer sammen!

**Klar til å bli en AI-kompetent helseprofesjonell? La oss begynne! 🎓**